app/login/actions.ts

```
@@ -27,7 +28,7 @@ export async function signInWithMagicLink(formData: FormData) {
       return `${protocol}://${process.env.VERCEL_URL}`
     }
     // For local development or custom deployments
-    return process.env.NEXT_PUBLIC_APP_URL || 'http://localhost:3000'
+    return env.get('NEXT_PUBLIC_APP_URL')!
```

If validation doesn't run or fails silently, these \! assertions will cause runtime crashes or pass bad values. Could you change these back to the original process.env access pattern?

---

No. I do not want to do that. The contract we are establishing is ...

- the env-var exists and was checked and has a meaningful value and are save to use (and do not need to be checked again before every use)
- if somebody puts a typo into the env-var name it will crash hard and fast

I think this is a clear and robust way to handle env-vars. WDYT?